src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -1312,6 +1531,20 @@ abstract class ManagedStrategy(
         }
     }
 
+    /**
+     * Current method context in a call stack.
+     */
+    sealed interface CallContext {
+        /**
+         * Indicates that current method is static.
+         */
+        data object StaticCallContext: CallContext
+
+        /**
+         * Indicates that method is called on the instance.
+         */
+        data class InstanceCallContext(val instance: Any): CallContext
```

Comment:
Why `data`?

Reply:
IDEA highlights this because it's a pattern (object in a sealed hierarchy)
![image](https://github.com/JetBrains/lincheck/assets/45040429/e5ebf808-a8e2-4419-82df-997975fa32b4)

Reply:
For me, it makes sense as it serves as a marker that this class is more like a structure. 
Also, it has a better toString(), which is useful for debugging